server/src/main/java/com/google/coffeehouse/storagehandler/StorageHandler.java

```diff
@@ -0,0 +1,304 @@
+public class StorageHandler {
+  public static final String NO_PRONOUNS = "No pronouns";
+  public static final String PERSON_DOES_NOT_EXIST = "This person does not exist in the database.";
+
+  public static final String CLUB_DOES_NOT_EXIST = "This club does not exist in the database.";
+
+  public static final String NO_AUTHOR = "No author";
+  public static final String NO_ISBN = "No ISBN";
+  public static final String BOOK_DOES_NOT_EXIST = "This book does not exist in the database.";
+
+  /**
+  * Creates and returns a {@link Person} with the result of a strong read from the database.
+  * This method builds a {@link Person} with a user's ID, email, nickname, and pronouns.
+  *
+  * @param  dbClient    the database client 
+  * @param  userId      the user ID string used to query and get a person's information
+  * @return person      the Person object built containing the person information
+  */
+  public static Person getPerson(DatabaseClient dbClient, String userId) {
+    String email = "";
+    String nickname = "";
+    String pronouns = "";
+    // long resultCount = StorageHandlerHelper.getPersonCount(dbClient, userId);
+    // if (resultCount == 0) {
+    //   throw new IllegalArgumentException(PERSON_DOES_NOT_EXIST);
+    // }
+    Struct row = 
+        dbClient
+            .singleUse()
+            .readRow(
+              "Persons",
+              Key.of(userId),
+              Arrays.asList("email", "nickname", "pronouns"));
+    if (row != null) {
+      email = row.getString(/*index=*/0);
+      nickname = row.getString(/*index=*/1);
+      if (!row.isNull(/*index=*/2) || !row.getString(/*index=*/2).isEmpty()) {
+        pronouns = row.getString(/*index=*/2);
+      }
+    } else {
+      throw new IllegalArgumentException(PERSON_DOES_NOT_EXIST);
+    }
+    Person.Builder personBuilder = Person.newBuilder(email, nickname);
+    // TODO: implement setting the userId field @JosephBushagour
+    if (!pronouns.isEmpty()) {
+      personBuilder.setPronouns(pronouns);
+    }
+    return personBuilder.build();
+  }
+
+  /**
+  * Creates and returns a {@link Book} with the result of a strong read from the database.
+  * This method builds a {@link Book} with a book's ID, author, ISBN, and title.
+  *
+  * @param  dbClient  the database client
+  * @param  bookId    the book ID string used to query and get a book's information
+  * @return book      the Book object built containing the book information
+  */
+  public static Book getBook(DatabaseClient dbClient, String bookId) {
+    String author = "";
+    String isbn = "";
+    String title = "";
+    // long resultCount = StorageHandlerHelper.getBookCount(dbClient, bookId);
+    // if (resultCount == 0) {
+    //   throw new IllegalArgumentException(BOOK_DOES_NOT_EXIST);
+    // }
+    Struct row = 
+        dbClient
+            .singleUse()
+            .readRow(
+              "Books",
+              Key.of(bookId),
+              Arrays.asList("author", "isbn", "title"));
```

Comment:
Similar to what Keaton pointed out above, I believe you could construct row inside the 'if (row != null) condition to potentially cut down on the number of variables you're using.

Reply:
I think you meant the construction of the different objects instead of the row (as that was what Keaton was mentioning), but I'll fix that!